site/app/models/GradingOrder.php

```diff
@@ -496,18 +496,30 @@ public function getSortedGradedGradeables() {
         );
         $gg_idx = [];
         $unsorted = [];
+        $items = [];
         foreach ($iter as $gg) {
-            $idx = $this->getSubmitterIndex($gg->getSubmitter());
-            //Should never happen, but better to be safe
-            if ($idx === false) {
-                $unsorted[] = $gg;
-            }
-            else {
-                $gg_idx[$idx] = $gg;
+            $items[$gg->getSubmitter()->getId()] = $gg;
+        }
+        $count = 0;
+
+        //Iterate through all sections and their submitters to find this one
+        foreach ($this->section_submitters as $name => $section) {
+            for ($i = 0; $i < count($section); $i++) {
+                $testSub = $section[$i]->getId();
+
+                //Found them
+                if (isset($items[$testSub])) {
+                    $gg_idx[$count] = $items[$testSub];
```

Comment:
Do we care about the gaps in `$gg_idx` between indexes, or can it just be purely sequential, so like `$gg_idx[] = $items[$testSub];` would work fine.